ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSpec.java

```diff
@@ -189,18 +194,18 @@ public static boolean allowReplacement(String currReplState, String replacementR
     }
 
     // First try to extract a long value from the strings, and compare them.
-    // If oldReplState is less-than or equal to newReplState, allow.
+    // If oldReplState is less-than newReplState, allow.
     long currReplStateLong = Long.parseLong(currReplState.replaceAll("\\D",""));
     long replacementReplStateLong = Long.parseLong(replacementReplState.replaceAll("\\D",""));
 
     if ((currReplStateLong != 0) || (replacementReplStateLong != 0)){
-      return ((currReplStateLong - replacementReplStateLong) <= 0);
+      return ((currReplStateLong - replacementReplStateLong) < 0);
     }
 
     // If the long value of both is 0, though, fall back to lexical comparison.
 
     // Lexical comparison according to locale will suffice for now, future might add more logic
-    return (collator.compare(currReplState.toLowerCase(), replacementReplState.toLowerCase()) <= 0);
+    return (collator.compare(currReplState.toLowerCase(), replacementReplState.toLowerCase()) < 0);
```

Comment:
I think, it shouldn't be a problem as we fallback to this compare function only if one of  the value is 0. Not sure, why is it there in first place. Need to check with Sushanth.